### ISSUE TITLE
[CI] test_torchintegration.py is partially disabled

### ISSUE THREAD

**Issue by jkwak-work** [shader-slang/slangpy#574]
One of the slangpy CI tests is partially disabled, `test_torchintegration.py`.
The name of the file implies that the intention is to test `pytorch` integration.

We need to investigate to decide to either:
- re-enable the test or
- delete the commented lines to avoid confusion.

The test was ported from SGL repo.
The following lines was active until it gets ported to slangpy repo.
The function call to `backward()` should be uncommented; or removed.
```
    # Should this work??
    # res.backward(torch.ones_like(res))
```
https://github.com/shader-slang/slangpy/blob/48231dc5b778329f5080fa3a79e2bbf8f295e860/slangpy/tests/slangpy_tests/test_torchintegration.py#L343-L344

**Comment by jkwak-work** [comment]
There is a Discord report about having a same problem:
- https://discord.com/channels/1303735196696445038/1412057410125697197

Raising the priority to P0 because this appears to be a blocker for people trying to migrate from slang-torch to slangpy.

**Comment by jkwak-work** [comment]
I don't understand what it means yet but Claude 4.5 says the following,
> The problem is clear: In the test_add_tensors function, res is created with torch.empty_like(a), which doesn't preserve the requires_grad property. When the    Slang function modifies it in-place, PyTorch doesn't build a gradient graph for it.

**Comment by jkwak-work** [comment]
I am going to ask @ccummingsNV  to review the comment above.
ClaudeCode suggested that with the following message:

>  Chris Cummings would be the ideal person to review this change since he:
>  - Authored the entire PyTorch autograd integration
>  - Worked extensively on the binding system including io_type logic
>  - Would understand both the autograd requirements and the binding architecture
>
>  Simon Kallweit would also be a good secondary reviewer as he implemented the current io_type property structure.

**Comment by ccummingsNV** [comment]
No - it's incorrect. The IO type defines whether the shader is outputting an RWTensor.

**Comment by jkwak-work** [comment]
Note that this file is a part of slangpy CI test.
The commented out line looks to be disabling a part of the testing and it may need to be reenabled.

**Comment by jkwak-work** [comment]
I am still learning about what the test is doing.
Chris told me that the test generates the following Slang shader,
```
[Differentiable]
void add_tensors(Tensor<float, 2> a, Tensor<float, 2> b, RWTensor<float, 2> res) {
    [MaxIters(8)]
    for (int i = 0; i < a.shape[0]; i++)
        [MaxIters(8)]
        for (int j = 0; j < a.shape[1]; j++)
            res.set({i, j}, a[i,j] + b[i, j]);
}
```
> I'd not expect anything to accumulate there, as none of those tensors contain grads

Here is my understanding.
Because the inputs such as `a` and `b` don't contain grads, `res` doesn't contain grads and we cannot call `backward()` when the tensor doesn't contain grads.

In order for it to call `backward()`, slangpy should have GradXXTensor objects such as GradInTensor or GradOutTensor. Because the pytorch code didn't use grad, Slang didn't emit any GradXXTensor objects.

I guess what LLM said is the Slang function signature should add `out` modifier to `RWTensor<float, 2> res`, but it is incorrect because the output shouldn't be on RWTensor but should be on GradXXTensor objects, which doesn't exist at all in the emitted code.

That makes me to believe that this test isn't supposed to work with `backward()`.

If we like to still test `backward()` function call, we may need to refactor to make them "accumulate".

**Comment by jkwak-work** [comment]
Based on what I have learned, I am lowering the priority from P1 to P2.
It doesn't seem urgent.

The next step is to check if we already have other CI tests that tests `backward()` call with pytorch.
If we have, we should remove the commented lines.

**Comment by jhelferty-nv** [comment]
*(Posted via Cursor (Claude Opus))*

The commented-out `res.backward()` lines have been removed in PR #928.

As ccummingsNV noted, backward on a non-differentiable RWTensor output isn't expected to accumulate gradients ("none of those tensors contain grads"). Multiple other tests already cover backward passes with proper differentiable outputs:

- `test_polynomial` / `test_polynomial_outparam` / `test_polynomials` (existing)
- `test_torch_autograd_workflows.py` (new in #928)

The remaining question -- whether backward *should* work on RWTensor outputs as a feature -- is a design decision beyond the scope of this test cleanup. Leaving this issue open for that discussion.
